docs/adrs/002-Retry_endpoint.md

```diff
@@ -0,0 +1,81 @@
+# Retry Endpoint
+
+Date: 2021-04-27
+
+The current requirement for the Batch/Bulk UI is to be able to retry jobs. Using a job id, it should be possible to get information from the database.
+
+The current implementation of retry is to run jobs using the `run_job` or `run_job_batch` endpoint. This is not adequate due to the following deficiencies:
+
+
+* Lack of book-keeping for the parent job and the child job relationship: Launching a new job will not update the relationship between those jobs
+* Lack of book-keeping for number of retries for a particular job / set of job inputs
+* Lack of ability to launch multiple jobs using the `run_job_batch` endpoint without creating a new parent job
+* Lack of ability to ensure that the proper catalog version /git commit of an app is used from the front end based on a tag, such as "beta/dev/release"
+* Lack of ability to specify which retries succeeded and which ones failed during submit time. 
```

Comment:
This needs more explanation -- what is "submit time"?

Reply:
Comment to be added into the document somehow: The job first needs to be submitted to ee2_runjob. It can fail there. Afterwards, it gets submitted to condor, it can fail there too. Currently those two happen at the same time, but they are supposed to happen in a thread at some point, so the current single point of submit time will become two separate points of submission.
Once the job begins running, it can fail at Run Time.

Reply:
Because it takes a long time to submit multiple jobs, and we want a list of job ids right away

Reply:
Ok, so you make the records in the DB and then start submitting to Condor, I see. That'll take a pretty big rewrite of `_run` though.